src/lib/bootstrap_controller/bootstrap_controller.ml

```diff
@@ -273,6 +306,41 @@ end = struct
       let root_sync_ledger =
         Root_sync_ledger.create ledger_db ~logger:t.logger ~trust_system
       in
+      (let%bind () = Ivar.read ask_best_tip_signal in
+       Logger.info logger
+         "Requesting peers for their best tip to eargly start bootstrap"
+         ~module_:__MODULE__ ~location:__LOC__ ;
+       match%bind
+         download_best_tip ~network ~logger ~trust_system ~verifier
+           initial_root_transition
+       with
```

Comment:
I think you should always do this before starting any of the other threads that pull information into the sync ledger. Under the new logic, you will always request the best tip first to get your initial logic. What happens here if you have already received something over gossip before you finish downloading the best tip from a peer? Would that cause a regression in the target?